Fixes all linting errors

erik-engstrom opened the pull request:
Fixes all linting errors in the Hyku app.
Closes ticket [#1](https://gitlab.com/notch8/intern-bravo-2021/-/issues/1)
Changes proposed in this pull request:
* Utilized Rubocop to identify linting errors
* Modified the Hyku app based on those linting errors
* Zero linting errors on Rubocop after making all modifications identified in Rubocop

@samvera/hyku-code-reviewers

guillermogb88 committed:
69 and 70 fixed in gem file

guillermogb88 committed:
fixed alphabetical order isssues in line 87 of the gem file

guillermogb88 committed:
gem file fixed

guillermogb88 committed:
works controller magic comment issue fixed

guillermogb88 committed:
resolve merge conflicts

guillermogb88 committed:
resolved merge conflicts

guillermogb88 committed:
resolved merge conflicts

Commit:
fix all linting errors

Commit:
fixed merge conflicts

labradford commented on app/controllers/hyrax/contact_form_controller.rb:
```diff
@@ -97,7 +97,7 @@ def inject_theme_views
           prepend_view_path(home_theme_view_path)
           yield
-          view_paths = original_paths
+          original_paths
```
Change this back to `view_paths=(original_paths)`

labradford replied:
this is calling a method called `view_paths=()`
https://apidock.com/rails/v2.3.8/ActionController/Base/view_paths%3D/class

erik-engstrom replied:
Excellent, thank you for the documentation on this as well. Rubocop was throwing a spacing and parenthesis error, maybe thinking it was trying to assign a variable instead of calling the method?

labradford replied:
yes, rubocop is wrong on this one. You can put the disable/enable comments around this line to bypass this message.

labradford replied:
this file still needs to be changed back to `view_paths=(original_paths)`

labradford commented on app/controllers/hyrax/homepage_controller.rb:
```diff
@@ -105,7 +105,7 @@ def inject_theme_views
           prepend_view_path(home_theme_view_path)
           yield
-          view_paths = original_paths
+          original_paths
```
Change this back to `view_paths=(original_paths)`

labradford commented on app/controllers/hyrax/pages_controller.rb:
```diff
@@ -99,7 +101,7 @@ def inject_theme_views
           prepend_view_path(home_theme_view_path)
           yield
-          view_paths=(original_paths)
+          original_paths
```
Change this back to `view_paths=(original_paths)`

labradford commented on .rubocop.yml:
```diff
@@ -111,10 +107,10 @@ RSpec/MessageSpies:
 
 RSpec/ExampleLength:
-  Max: 16
+  Max: 31
```
I don't think it is a good idea to change the project's rubocop settings. We would need to check with the Hyku project owner before making changes to this file.

erik-engstrom replied:
Thank you so much for this feedback. I wasn't quite sure if I could just change the specs. This definitely answers that question. So we'd just put the # disable rubocop line length limit comment in the files where the error is?

labradford replied:
yes, or reformat the line to move it into multiple lines

labradford commented on .rubocop.yml:
```diff
@@ -111,10 +107,10 @@ RSpec/MessageSpies:
 
 RSpec/NestedGroups:
-  Max: 4
+  Max: 6
```
Verify with the project owner before making changes to the rubocop settings for Hyku.

labradford reviewed (changes requested):
I left some changes for you. Also, I would want to make sure that the changes you made to rspec files did not break the tests before merging those changes.

erik-engstrom committed:
Update homepage_controller.rb

erik-engstrom committed:
Update pages_controller.rb

Added back the method view_paths

labradford commented on app/models/featured_work.rb:
```diff
@@ -1,21 +1,24 @@
 
-  default_scope { order(:order) }
+class FeaturedWork < ActiveRecord
+  class Base
```
I'm not sure this is correct here

labradford replied:
Base is not a module inside of FeaturedWork

labradford commented on app/controllers/hyku/invitations_controller.rb:
```diff
@@ -1,33 +1,33 @@
-      hyrax.admin_users_path
-    end
+  class InvitationsController < Devise
+    class InvitationsController
```
This formatting looks wrong here. Not sure what the rubocop error is, but you should be inheriting from this class, not creating a submodule.

erik-engstrom committed:
Update on rubocop.yml

Reverted back to the owners example length max = 16  and nested groups max = 4 specifications.

erik-engstrom committed:
Update invitations_controller.rb

erik-engstrom committed:
Update featured_work.rb

Reverted back to class FeaturedWork < ActiveRecord::Base

labradford commented on app/models/featured_work.rb:
```diff
@@ -1,21 +1,23 @@
-  FEATURE_LIMIT = 6
-  validate :count_within_limit, on: :create
-  validates :order, inclusion: { in: proc { 0..FEATURE_LIMIT } }
+    FEATURE_LIMIT = 6
```
this spacing looks off to me

labradford commented on lib/tasks/tenantize_task.rake:
```diff
@@ -1,6 +1,5 @@
 # frozen_string_literal: true
 
-# rubocop:disable Metrics/LineLength
```
if you delete this line, you also need to remove the `rubocop:enable Metrics/LineLength` on line 20

labradford commented on spec/controllers/hyrax/content_blocks_controller_spec.rb:
```diff
@@ -68,7 +70,6 @@
 
       # Copied from Hyrax v2.9.0 to add home_text content block - Adding themes
       it "updates the home text" do
-        # rubocop:disable Metrics/LineLength
```
you also need to remove the `rubocop:enable Metrics/LineLength` on line 84

labradford commented on spec/features/appearance_theme_spec.rb:
```diff
@@ -65,7 +66,6 @@
       select('Gallery view', from: 'Search Results Page Theme')
-      # rubocop:disable Metrics/LineLength
       expect(page).to have_content('This will select a default view for the search results page. Users can select their preferred views on the search results page that will override this selection')
       # rubocop:enable Metrics/LineLength
```
you need to remove the enable block when you remove disable blocks

labradford commented on spec/features/create_generic_work_spec.rb:
```diff
@@ -39,7 +39,6 @@
       login_as user
     end
 
-    # rubocop:disable RSpec/ExampleLength
```
you also need to remove the `rubocop:enable RSpec/ExampleLength on line 80

labradford commented on spec/features/create_generic_work_spec.rb:
```diff
@@ -68,7 +67,6 @@
       choose('generic_work_visibility_open')
-      # rubocop:disable Metrics/LineLength
       expect(page).to have_content('Please note, making something visible to the world (i.e. marking this as Public) may be viewed as publishing which could impact your ability to')
       # rubocop:enable Metrics/LineLength
```
you need to remove this line

labradford commented on spec/features/create_image_spec.rb:
```diff
@@ -67,7 +66,6 @@
       choose('image_visibility_open')
-      # rubocop:disable Metrics/LineLength
       expect(page).to have_content('Please note, making something visible to the world (i.e. marking this as Public) may be viewed as publishing which could impact your ability to')
       # rubocop:enable Metrics/LineLength
```
you need to remove this line

labradford commented on spec/features/create_image_spec.rb:
```diff
@@ -38,7 +38,6 @@
       login_as user
     end
 
-    # rubocop:disable RSpec/ExampleLength
```
You need to remove the `rubocop:enable RSpec/ExampleLength` on line 79

labradford commented on spec/features/institutional_repository_theme_spec.rb:
```diff
@@ -71,4 +71,4 @@
   end
-end
\ No newline at end of file
+end
```
looks like it still needs an empty line at the end of file

labradford commented on spec/support/factory_bot.rb:
```diff
@@ -1,3 +1,5 @@
-end
\ No newline at end of file
+  config.include FactoryBot::Syntax::Methods
+end
```
looks like it still needs a new line at the end of the file

labradford commented on spec/views/hyrax/admin/appearances/show.html.erb_spec.rb:
```diff
@@ -25,7 +25,6 @@
       # default collection image
-      # rubocop:disable Metrics/LineLength
       assert_select "input#admin_appearance_default_collection_image[name=?]", "admin_appearance[default_collection_image]"
       # rubocop:enable Metrics/LineLength
```
you need to remove this line

labradford commented on app/controllers/hyku/invitations_controller.rb:
```diff
@@ -2,32 +2,31 @@
-    def after_invite_path_for(_resource)
-      hyrax.admin_users_path
-    end
+      # For devise_invitable, specify post-invite path to be 'Manage Users' form
```
the spacing looks off to me in this file

labradford replied:
was this just linting on this file? I don't recall it being a part of the theming work?

Commit:
rubocop fixes

Commit:
fixes all the rubocop errors

labradford commented on app/controllers/hyku/invitations_controller.rb:
```diff
@@ -26,7 +26,7 @@ def create
 
       # Override destination as this was a success either way
-      if is_flashing_format? && resource.invitation_sent_at
+      if is_flashing_format? && resource.invitation_sent_at # rubocop:disable Style/IfUnlessModifier
```
Still confused as to why this change is being made. It's not part of the theming work.

labradford replied:
Because this is already in Hyku without the disable comment, I would remove it.

labradford commented on app/controllers/concerns/hyrax/works_controller_behavior.rb:
```diff
@@ -353,11 +348,7 @@ def inject_show_theme_views
           prepend_view_path(show_theme_view_path)
           yield
-<<<<<<< HEAD
           original_paths
```
this should be `view_paths=(original_paths)`

labradford reviewed (changes requested):
Just two more small changes and we can get this merged.

Commit:
Fixes rubocop
